### PR TITLE
platform: Introduce a new platform API

### PR DESCRIPTION
**Pull request by tomchy**
Introduce a possibility to switch between old and new platform API as well as make the dry run functionality optional.

### CODE REVIEW

**Review by maciejpietras — CHANGES_REQUESTED** [review]
Please modify licenses in all modified files according to https://nordicsemi.atlassian.net/wiki/spaces/SPG/pages/66434179/DFU+Component+licenses+overview

File: include/suit_platform.h [hunk 1]
```diff
@@ -10,6 +10,10 @@
 #include "suit_types.h"
 #include <stdint.h>
 
+#ifdef SUIT_PLATFORM_LEGACY_API_SUPPORT
```
Comment (rafp83) [inline include/suit_platform.h:13]: Shouldn't that be "CONFIG_SUIT_PLATFORM_LEGACY_API_SUPPORT"?

File: include/suit_platform.h [hunk 2]
```diff
@@ -27,109 +31,123 @@
...
-		suit_component_t image_handle);
+/** Release loaded component properties and handles assigned to them.
+ *
+ *  @param[in]  component_handle  The platform-specific component handle value.
+ */
+int suit_plat_release_component_handle(suit_component_t handle);
 
-/** Check whether the given slot is active for the given component. */
-int suit_plat_check_slot(suit_component_t component_handle, unsigned int slot);
+#ifndef SUIT_PLATFORM_LEGACY_API_SUPPORT
```
Comment (rafp83) [inline include/suit_platform.h:75]: Shouldn't that be "CONFIG_SUIT_PLATFORM_LEGACY_API_SUPPORT"?

File: include/suit_platform.h (hunk 1 above)
Comment (tomchy) [reply include/suit_platform.h:13]: It depends. I chose to go the "ZCBOR" way, so to use regular defines and map Kconfigs to them inside CMake.

File: include/suit_platform.h [hunk 3]
```diff
@@ -27,109 +31,123 @@
...
+/** Check the provided payload against the provided digest. */
+int suit_plat_check_image_match(suit_component_t handle,
+		enum suit_cose_alg alg_id, struct zcbor_string *digest,
+		size_t image_size);
+#endif /* !SUIT_PLATFORM_LEGACY_API_SUPPORT */
 
+/** Check the provided payload against the component value. */
```
Comment (rafp83) [inline include/suit_platform.h:82]: There's some inconsistency in the way functions are documented. Why not all functions have their arguments documented?

**Review by rafp83 — APPROVED** [review]

**Review by robertstypa — APPROVED** [review]

File: include/suit_platform.h (hunk 3 above)
Comment (tomchy) [reply include/suit_platform.h]: Docs inside the `suit_platform.h` added.

**Comment by tomchy** [comment]
PR with the west.yml update: https://github.com/nrfconnect/sdk-nrf-next/pull/1200

**Pull request merged**